src/core/storage/fileio/fs_utils.cpp

```diff
@@ -41,7 +41,7 @@ EXPORT std::string make_canonical_path(const std::string& path) {
     } else {
       return fs::canonical(fs::absolute(p)).string();
     }
-  } catch (fs::filesystem_error e) {
+  } catch (fs::filesystem_error& e) {
```

Comment:
These should be changed to const type&.  This changes the behavior -- objects passed by value will not match a type& catch statement.